### PR TITLE
fix(FrameworkConfiguration): move legal t&c text when installing

### PR DESCRIPTION
**Pull request by ahoskins**
It is a legal requirement to show the terms and conditions notice on the same page as the action that deploys the service. This change does this by moving the T&C link to the new framework review screen. This will only show when deploying from catalog, not when editing the framework. This change also removes the T&C link from the package detail page.

**Most frameworks have preinstall notes, so we append the T&C to the end:**
image 2017-11-28 at 1 50 09 pm

**If it doesn't have preinstall notes, we just show the T&C (faked example, couldn't find a service without preinstall notes):**
image 2017-11-28 at 1 56 01 pm

**The T&C link is now removed from package detail:**
image 2017-11-28 at 2 19 17 pm

Closes DCOS-19359.

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by ahoskins** [comment]
@leemunroe looping you in since you are creator of the Jira.

File: src/js/structs/UniversePackage.js [hunk 1]
```diff
@@ -68,7 +68,13 @@ class UniversePackage extends Item {
   }
 
   getPreInstallNotes() {
-    return this.get("preInstallNotes");
+    // punctuate the end if not present
```
Comment (ahoskins) [inline src/js/structs/UniversePackage.js]: I'm adding this because there are some packages (for example [cassandra](https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/C/cassandra/29/package.json#L19)) don't end the preinstall notes with a period. The proper fix would be for the package itself to not do this, but I think it's worth the patch in the UI code. If I don't add this, then when appending the T&C it lacks a period, which is not nice.

**Comment by ahoskins** [comment]
Alfreddo test this please

File: src/js/structs/UniversePackage.js (hunk 1 above)
Comment (weblancaster) [reply src/js/structs/UniversePackage.js]: It feels like this should be inside a util instead the struct

Comment (ahoskins) [reply src/js/structs/UniversePackage.js]: Yeah true, I can move into string util.

File: src/js/utils/StringUtil.js [hunk 2]
```diff
@@ -84,6 +84,18 @@ const StringUtil = {
     return string;
   },
 
+  punctuate(string) {
```
Comment (nLight) [inline src/js/utils/StringUtil.js]: Would be great to see a test or two for this util :)

File: src/js/utils/StringUtil.js [hunk 3]
```diff
@@ -84,6 +84,18 @@ const StringUtil = {
...
+    if (typeof string !== "string") {
+      return null;
```
Comment (orlandohohmeier) [inline src/js/utils/StringUtil.js]: 🎨  I'd return an empty string to not change the return type.

**Comment by ahoskins** [comment]
@orlandohohmeier @nLight I've added tests to `punctuate` and change to return `""` in error case. Should be good to 🚢

File: src/js/utils/__tests__/StringUtil-test.js [hunk 4]
```diff
@@ -206,6 +206,30 @@ describe("StringUtil", function() {
...
+        "butterfly is nice."
+      );
+    });
+
+    it("does not further punctuate a sentence with a period", function() {
+      expect(StringUtil.punctuate("butterfly is nice.")).toEqual(
+        "butterfly is nice."
+      );
+    });
+  });
```
Comment (juliangieseke) [inline src/js/utils/__tests__/StringUtil-test.js]: I'd love to see some edge cases here. 
`StringUtil.punctuate("");`
`StringUtil.punctuate(new Array());`
`StringUtil.punctuate(null);`
`StringUtil.punctuate(1);`

**Comment by ahoskins** [comment]
@juliangieseke I've added a couple more test cases for weird inputs 👍

**Comment by leemunroe** [comment]
@ahoskins FYI the link color in this message should inherit the color of the message text i.e. should be brown not purple. Similar how we do it for other messages and alerts e.g. this danger message

image

**Comment by ahoskins** [comment]
@leemunroe Cool thanks, I didn't notice that. Will push a fix for this.

File: src/js/components/FrameworkConfiguration.js [hunk 5]
```diff
@@ -198,34 +198,55 @@ class FrameworkConfiguration extends Component {
...
-        "<strong>Preinstall Notes: </strong>" + preInstallNotesParsed.__html;
+    if (preInstallNotes) {
+      const message = StringUtil.parseMarkdown(preInstallNotes);
+      message.__html = `<strong>Preinstall Notes: </strong>${message.__html} ${this.getTermsAndConditions().__html}`;
 
-      preinstall = (
+      return (
         <div
-          dangerouslySetInnerHTML={preInstallNotesParsed}
+          dangerouslySetInnerHTML={message}
```
Comment (juliangieseke) [inline src/js/components/FrameworkConfiguration.js]: any chance to get rid of this?

File: src/js/components/FrameworkConfiguration.js [hunk 6]
```diff
@@ -198,34 +198,55 @@ class FrameworkConfiguration extends Component {
...
           className="pre-install-notes message message-warning"
         />
       );
     }
 
+    return (
+      <div
+        dangerouslySetInnerHTML={this.getTermsAndConditions()}
```
Comment (juliangieseke) [inline src/js/components/FrameworkConfiguration.js]: same here

File: src/js/components/FrameworkConfiguration.js [hunk 7]
```diff
@@ -235,7 +256,7 @@ class FrameworkConfiguration extends Component {
       <div className="flex-item-grow-1">
         <div className="container container-wide">
           {errorsAlert}
-          {preinstall}
+          {warningMessage}
```
Comment (juliangieseke) [inline src/js/components/FrameworkConfiguration.js]: why did you decide to change this? is it really a "warning"? wouldn't it be possible to split them into two? 
```
{preinstall}
{tandc}
```

File: src/js/components/FrameworkConfiguration.js (hunk 5 above)
Comment (nLight) [reply src/js/components/FrameworkConfiguration.js]: I think the answer is "no" since those messages can contain html we want to render.

Comment (ahoskins) [reply src/js/components/FrameworkConfiguration.js]: Yeah, this is a pattern we follow [elsewhere](https://github.com/dcos/dcos-ui/blob/master/src/js/pages/catalog/PackageDetailTab.js#L429) after parsing markdown. It's ok - I don't feel too dangerous doing this 😉

File: src/js/components/FrameworkConfiguration.js (hunk 7 above)
Comment (ahoskins) [reply src/js/components/FrameworkConfiguration.js]: It must be as one because the content is part of the same paragraph (the preinstall and t&c are concat)
